unity-renderer/Assets/Scripts/MainScripts/DCL/WorldRuntime/KernelCommunication/BinaryMessage/KernelBinaryMessageSerializer.cs

```diff
@@ -0,0 +1,19 @@
+using DCL.CRDT;
+
+namespace KernelCommunication
+{
+    public static class KernelBinaryMessageSerializer
+    {
+        public static void Serialize(BinaryWriter binaryWriter, CRDTMessage message)
+        {
+            //sizeof(messageHeader) + sizeof(componentHeader) + dataLength
+            int messageLength = BinaryMessageConstants.CRDT_MESSAGE_BASE_HEADER_LENGTH + (((byte[])message.data)?.Length ?? 0);
+            var type = message.data != null ? KernelBinaryMessageType.PUT_COMPONENT : KernelBinaryMessageType.DELETE_COMPONENT;
```

Comment:
I find a bit weird that we use the status of `message.data` to know whether the message is a PUT or a DELETE.

Let's at least wrap it in a `IsPUTMessage()`  method somewhere with a comment to make it easier to maintain.